ui/src/App.js

```diff
@@ -133,64 +135,85 @@ class App extends Component {
     datasetApi.datasetGet(datasetCallback);
   }
 
-  handleSearch(searchResults) {
-    let extraFacetEsFieldNames = this.state.extraFacetEsFieldNames;
-    for (let i = 0; i < searchResults.length; i++) {
-      if (searchResults[i].facetValue == "") {
-        extraFacetEsFieldNames.push(searchResults[i].esFieldName);
-      }
-    }
-
-    let filterArray = this.filterMapToArray(this.filterMap);
-    this.setState({
-      extraFacetEsFieldNames: extraFacetEsFieldNames,
-      filter: filterArray
+  getFacetMap(facets) {
+    var facetMap = new Map();
+    facets.forEach(function(facet) {
+      facetMap.set(facet.es_field_name, facet);
     });
-    this.facetsApi.facetsGet(
-      {
-        filter: filterArray,
-        extraFacets: extraFacetEsFieldNames
-      },
-      this.facetsCallback
-    );
+    return facetMap;
+  }
+
+  handleSearchBoxChange(selectedOptions, action) {
+    if (action.action == "clear") {
+      // x on right of search box was clicked.
+      this.setState({ selectedFacetValues: new Map() });
+      this.facetsApi.facetsGet(
+        {
+          filter: this.filterMapToArray(new Map()),
+          extraFacets: this.state.extraFacetEsFieldNames
+        },
+        this.facetsCallback
+      );
+    } else if (action.action == "remove-value") {
+      // chip x was clicked.
+      // selectedOptions contains remaining selected facet values (ie remaining chips).
```

Comment:
Delete 159-160, the comments are no longer relevant because we're not using selectedOptions